PagingWithNetworkSample/lib/src/main/java/com/android/example/paging/pagingwithnetwork/reddit/paging/SynchronousRemoteStates.kt

```diff
@@ -0,0 +1,141 @@
+package com.android.example.paging.pagingwithnetwork.reddit.paging
+
+import androidx.paging.CombinedLoadStates
+import androidx.paging.LoadState
+import androidx.paging.LoadState.*
+import androidx.paging.LoadStates
+import com.android.example.paging.pagingwithnetwork.reddit.paging.SynchronousRemoteState.*
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.scan
+import kotlin.Error
+
+/**
+ * Track the combined [LoadState] of [RemoteMediator] and [PagingSource], so that each load type
+ * is only set to [NotLoading] when [RemoteMediator] load is applied on presenter-side.
+ */
+private class SynchronousRemoteStates {
```

Comment:
Naming is confusing here - asSynchronousRemoteStates doesn't return this, or the SynchronousRemoteState object.

Suggest:

SynchronousRemoteStates -> LoadStateSimplifier (Serializer, Merger, Manager, something like that)
SynchronousRemoteState -> MergedLoadState

Reply:
Done